### PR TITLE
Add a new button for the summarization process. Multi-user support

### PR DESCRIPTION
**Pull request by abrecko**
Edit:
- Adding a button to summarize
- Multi-user support
- Stop/reset process options
- Possibility to change the input file

### CODE REVIEW

File: src/gradio/app/main.py [hunk 1]
```diff
@@ -20,40 +20,54 @@
...
+        summary_tasks[session_id] = task
+        output_text = await task
+        logging.info("Summarization completed.")
+        return output_text
+    except Exception as e:
+        logging.error(f"Failed during processing: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to process PDF: {str(e)}")
+    except asyncio.CancelledError:
+        logging.warning(f"Summarization was cancelled for session {session_id}.")
+        raise HTTPException(
```
Comment (andrejridzik) [inline src/gradio/app/main.py:65]: Replace raising error by either `gr.Info("Summarization was cancelled")` or just simply omit the exception

File: src/gradio/app/main.py [hunk 2]
```diff
@@ -112,64 +126,70 @@ async def generate_post_async(
...
+            generated_post = await llm_task
+            return generated_post
+        except Exception as e:
+            logging.error(f"Failed during processing: {str(e)}")
+            raise HTTPException(
+                status_code=500, detail=f"Failed to process PDF: {str(e)}"
+            )
+        except asyncio.CancelledError:
+            logging.warning(f"Summarization was cancelled for session {session_id}.")
+            raise HTTPException(status_code=400, detail="Summarization was cancelled")
```
Comment (andrejridzik) [inline src/gradio/app/main.py:162]: Replace raising error by either gr.Info("Text generation was cancelled") or just simply omit the exception

File: src/gradio/app/main.py [hunk 3]
```diff
@@ -112,64 +126,70 @@ async def generate_post_async(
...
+        try:
+            loop = asyncio.get_running_loop()
+            llm_task = loop.run_in_executor(None, llm.invoke, post_text)
+            llm_tasks[session_id] = llm_task
+            generated_post = await llm_task
+            return generated_post
+        except Exception as e:
+            logging.error(f"Failed during processing: {str(e)}")
+            raise HTTPException(
+                status_code=500, detail=f"Failed to process PDF: {str(e)}"
```
Comment (andrejridzik) [inline src/gradio/app/main.py:158]: Incorrect message, should be related to text generation

File: src/gradio/app/main.py [hunk 4]
```diff
@@ -112,64 +126,70 @@ async def generate_post_async(
...
+            )
+        except asyncio.CancelledError:
+            logging.warning(f"Summarization was cancelled for session {session_id}.")
```
Comment (andrejridzik) [inline src/gradio/app/main.py:161]: Incorrect message, should be related to text generation

File: src/gradio/app/main.py [hunk 5]
```diff
@@ -112,64 +126,70 @@ async def generate_post_async(
...
+            raise HTTPException(status_code=400, detail="Summarization was cancelled")
 
 
 def stop_llm(session_id):
     global llm_tasks
     task = llm_tasks.get(session_id)
     if task is not None:
-        task.cancel()
-        return "LLM processing has been stopped."
+        logging.info(f"Summarization task for session {session_id} was cancelled.")
```
Comment (andrejridzik) [inline src/gradio/app/main.py:169]: Incorrect message, should be related to text generation

File: src/gradio/app/main.py [hunk 6]
```diff
@@ -112,64 +126,70 @@ async def generate_post_async(
...
+        summary_tasks.pop(session_id, None)
+        raise HTTPException(status_code=400, detail="Summarization was cancelled")
     else:
-        logging.info(f"No LLM process was found running for session {session_id}.")
-        return "No active LLM process to stop."
+        logging.info(
+            f"No summarization process was found running for session {session_id}."
```
Comment (andrejridzik) [inline src/gradio/app/main.py:174]: Incorrect message, should be related to text generation

File: src/gradio/app/main.py [hunk 7]
```diff
@@ -184,61 +204,70 @@ async def reset_summarization(session_id):
...
+        summary_output = gr.Textbox(label="Summary of the PDF", lines=6)
+        start_summarization_button = gr.Button("Start summarization")
+        reset_button = gr.Button("Reset")
+
+    with second_part:
+        gr.Markdown("## Social Media Post Panel")
+
+        audience_input = gr.Dropdown(
+            [
+                "",
```
Comment (andrejridzik) [inline src/gradio/app/main.py:222]: Add this to all options (if it works)

File: src/gradio/app/main.py [hunk 8]
```diff
@@ -20,40 +20,54 @@
...
+        logging.error(f"Failed during processing: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to process PDF: {str(e)}")
+    except asyncio.CancelledError:
+        logging.warning(f"Summarization was cancelled for session {session_id}.")
+        raise HTTPException(
+            status_code=400, detail="Summarization was cancelled"
+        )  # gr.info()
 
 
+# delete input from UI
```
Comment (andrejridzik) [inline src/gradio/app/main.py:70]: You can delete this line

**Review by andrejridzik — CHANGES_REQUESTED** [review]

File: src/gradio/app/main.py (hunk 2 above)
Comment (abrecko) [reply src/gradio/app/main.py]: `gr.info()` does not exist and cannot be used to inform the user. Do you mean print asyncio.CancelledError?

File: src/gradio/app/main.py (hunk 1 above)
Comment (abrecko) [reply src/gradio/app/main.py]: `gr.info()` does not exist and cannot be used to inform the user. Do you mean print asyncio.CancelledError?

File: src/gradio/app/main.py (hunk 2 above)
Comment (andrejridzik) [reply src/gradio/app/main.py]: Same as above

**Review by andrejridzik — APPROVED** [review]

**Pull request merged**